src/test/functional/steps/court-addresses.ts

```diff
@@ -0,0 +1,156 @@
+import {Then, Given, When} from 'cucumber';
+import {expect} from 'chai';
+
+import * as I from '../utlis/puppeteer.util';
+
+Then('I click the Addresses tab', async () => {
+  const selector = '#tab_addresses';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.click(selector);
+});
+
+Given('I will make sure to clear all entries of the primary address', async () => {
+  const selectorPrimaryAddressLines = '#primaryAddressLines';
+  const selectorPrimaryAddressLinesExist = await I.checkElement(selectorPrimaryAddressLines);
+  expect(selectorPrimaryAddressLinesExist).equal(true);
+  await I.clearField(selectorPrimaryAddressLines);
+
+  const selectorPrimaryAddressLinesWelsh = '#primaryAddressWelsh';
+  const selectorPrimaryAddressLinesWelshExist = await I.checkElement(selectorPrimaryAddressLinesWelsh);
+  expect(selectorPrimaryAddressLinesWelshExist).equal(true);
+  await I.clearField(selectorPrimaryAddressLinesWelsh);
+
+  const selectorPrimaryAddressTown = '#primaryAddressTown';
+  const selectorPrimaryAddressTownExist = await I.checkElement(selectorPrimaryAddressTown);
+  expect(selectorPrimaryAddressTownExist).equal(true);
+  await I.clearField(selectorPrimaryAddressTown);
+
+  const selectorPrimaryAddressTownWelsh = '#primaryAddressTownWelsh';
+  const selectorPrimaryAddressTownWelshExist = await I.checkElement(selectorPrimaryAddressTownWelsh);
+  expect(selectorPrimaryAddressTownWelshExist).equal(true);
+  await I.clearField(selectorPrimaryAddressTownWelsh);
+
+  const selectorPrimaryAddressPostcode = '#primaryAddressPostcode';
+  const selectorPrimaryAddressPostcodeExist = await I.checkElement(selectorPrimaryAddressPostcode);
+  expect(selectorPrimaryAddressPostcodeExist).equal(true);
+  await I.clearField(selectorPrimaryAddressPostcode);
+});
+
+When('I select the Address Type {string}', async (addressType: string) => {
+  const selector = '#primaryAddressType';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.selectItem(selector,addressType);
+});
+
+Then('I enter court {string} in the Address textbox', async (address: string) => {
+  const selector = '#primaryAddressLines';
```

Comment:
I think the lines here and for a few Then's below are duplicated with the exception of the string being used to set the element, like above can probably create a method for this, like: 

```
async function populateField(fieldElement: string, value: string) {
  expect(await I.checkElement(selector)).equal(true);
  await I.setElementValueForInputField(fieldElement, value);
}
```

```
Then('I enter {string} in the Address Welsh textbox', async (welshAddress: string) => {
 await populateField('#primaryAddressWelsh', welshAddress);
});

Then('I enter {string} in the Town textbox', async (town: string) => {
  const selector = '#primaryAddressTown';
 await populateField('#primaryAddressTown', town);
});
```

Reply:
I've created this method.